File: cache/cache-annotation-processor/src/test/java/ru/tinkoff/kora/cache/annotation/processor/testcache/DummyCache21.java
package ru.tinkoff.kora.cache.annotation.processor.testcache;

import ru.tinkoff.kora.cache.annotation.Cache;
import ru.tinkoff.kora.cache.caffeine.CaffeineCache;

import java.math.BigDecimal;

@Cache("dummy21")
public interface DummyCache21 extends CaffeineCache<DummyCache21.Key, String> {

    record Key(String k1, BigDecimal k2) {}
}

File: cache/cache-symbol-processor/src/test/kotlin/ru/tinkoff/kora/cache/symbol/processor/testcache/DummyCache21.kt
package ru.tinkoff.kora.cache.symbol.processor.testcache

import ru.tinkoff.kora.cache.annotation.Cache
import ru.tinkoff.kora.cache.caffeine.CaffeineCache
import java.math.BigDecimal

@Cache("dummy21")
interface DummyCache21 : CaffeineCache<DummyCache21.Key, String> {

    data class Key(val k1: String, val k2: BigDecimal)
}
